lib/SILGen/SILGenPoly.cpp

```diff
@@ -3746,8 +3761,8 @@ SILGenModule::getOrCreateAutoDiffAssociatedFunctionThunk(
 
   SmallVector<SILValue, 8> directResults;
   extractAllElements(apply, thunkSGF.B, directResults);
-  auto linearMap = directResults.back();
-  auto linearMapFnType = linearMap->getType().castTo<SILFunctionType>();
+  auto linearMap = ManagedValue::forBorrowedObjectRValue(directResults.back());
```

Comment:
You aren’t using any SILGen memory management in this thunk. This should be ‘ManagedValue::forUnmanaged’ or something close (I forgot the exact name).

Reply:
Will address shortly in a follow-up.
(I plan to enable ownership for AD associated function thunks soon.)